### PR TITLE
Feature/kak/travel time by mode#837

### LINKED ISSUE #850: Display list view summary of per-mode travel times and distances
First and last sub-tasks for #837.

Add to the itinerary object summary information, per mode, of travel times, distances, and number of transfers. Display on list view. Combine transit mode types to a single mode.

### LINKED ISSUE #851: Update dependencies
Update JavaScript dependencies in `package.json` and `bower.json`. Upgrade node/npm versions.

Fix issue with Browserify build of turf.js:
```
Cannot find module '/opt/app/src/node_modules/@turf/point-on-line/node_modules/@turf/helpers' from '/opt/app/src'
```
Package has changed sub-module installation paths.

### LINKED ISSUE #853: Do not display per-mode summary if there is only one mode
Sub-task of #837.

For single-mode trips, only display total travel time and distance (and not the per-mode breakdown).

### LINKED ISSUE #854: Only display mode summary for modes used over a given distance
Sub-task of #837.

Set a minimum threshold for showing a mode type in the summary (i.e., no travel time for a walk under 100 ft).

### PR DESCRIPTION
**Pull request by flibbertigibbet**
## Overview

Updates packages to fix JavaScript build and tests. Adds trip per-mode summary information to itinerary list view cards.

### Demo

image

### Notes

Needs styling work. The mode icon class has been applied, but the icon does not appear.

## Testing Instructions

 * Reprovision app server to pick up new dependencies
 * Plan a trip
 * Should get summary information in sidebar for per-mode travel times, distance, and transfer count
 * Short walks (less than a city block or so) should not display in the summary
 * Summary should only display with at least two modes

Closes #851 
Closes #850 
Closes #854
Closes #853

### CODE REVIEW

**Comment by flibbertigibbet** [comment]
Messed with the styling a bit to get the mode icons to show up:

image

**Comment by flibbertigibbet** [comment]
New and improved single-mode summary:

image

**Comment by lederer** [comment]
LGTM from the design side. As mentioned, I'll follow up with a styling PR.

File: src/app/scripts/cac/routing/cac-routing-itinerary.js [hunk 1]
```diff
@@ -107,6 +121,32 @@ CAC.Routing.Itinerary = (function ($, cartodb, L, _, moment, Geocoder, Utils) {
...
+    function getModeSummaries(legs) {
+        // minimum length of travel via a given mode for it to show up in the summary
+        // (about a city block)
+        var MIN_MODE_LENGTH_METERS = 120;
+
+        return _.chain(legs).groupBy(function(leg) {
+            return leg.transitLeg ? 'TRANSIT' : leg.mode;
+        }).mapValues(function(modeLegs) {
+            var dist = _.map(modeLegs, 'distance').reduce(function(sum, d) { return sum + d; });
+            var time = _.map(modeLegs, 'duration').reduce(function(sum, d) { return sum + d; });
```
Comment (KlaasH) [inline src/app/scripts/cac/routing/cac-routing-itinerary.js]: Lodash, in its extensive and sometimes murky bag of tricks, has `sumBy`.  So these could be 
```
var dist = _.sumBy(modeLegs, 'distance')
var time = _.sumBy(modeLegs, 'duration')
```

Comment (flibbertigibbet) [inline src/app/scripts/cac/routing/cac-routing-itinerary.js]: Nice find. It was just added in version 4. We have some existing `sum` references that should be replaced for consistency then, if we're going to switch to the convenience function.

**Pull request merged**